### PR TITLE
cmd: refactor building logic into a helper

### PR DESCRIPTION
**Pull request by polinasok**
This refactoring reduces code duplication and makes the helper available to be used by dlv-dap command in the future.

### CODE REVIEW

**Comment by polinasok** [comment]
I am looking into the test failure.

File: cmd/dlv/cmds/commands.go [hunk 1]
```diff
@@ -472,18 +472,30 @@ func dapCmd(cmd *cobra.Command, args []string) {
 	os.Exit(status)
 }
 
+func buildBinary(cmd *cobra.Command, args []string, isTest bool) string {
```
Comment (hyangah) [inline cmd/dlv/cmds/commands.go:475]: How about making this to return (string, bool) or (string, error)? 
https://github.com/golang/go/wiki/CodeReviewComments#in-band-errors

Comment (polinasok) [reply cmd/dlv/cmds/commands.go]: Good point. Done.

**Review by aarzilli — APPROVED** [review]

**Review by derekparker — APPROVED** [review]
LGTM

**Pull request merged**